app/controllers/profiles_controller.rb

```diff
@@ -1,4 +1,23 @@
 class ProfilesController < ApplicationController
+  
   def new
+    @profile = Profile.new
   end
+
+  def create
```

Comment:
The way you do this opens this action to non-authenticated users.

Reply:
I would look into using a `before_action` and make sure the user is authenticated first. You might argue that you regulate this on the view, but IMO it's not enough.

Reply:
done.